Dockerfile

```diff
@@ -31,35 +31,49 @@ RUN pip install --target=/dependencies playwright~=1.27.1 \
 # Final image stage
 FROM python:3.10-slim
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
-    libssl1.1 \
-    libxslt1.1 \
-    # For pdftohtml
-    poppler-utils \
-    zlib1g \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
 
+RUN set -ex; \
+    apt-get update && apt-get install -y --no-install-recommends \
+        gosu \
+        libssl1.1 \
+        libxslt1.1 \
+        # For pdftohtml
+        poppler-utils \
+        zlib1g && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*; \
+    useradd -u 911 -U -d /datastore -s /bin/false changedetection && \
+    usermod -G users changedetection; \
+    mkdir -p /datastore
 
 # https://stackoverflow.com/questions/58701233/docker-logs-erroneously-appears-empty-until-container-stops
 ENV PYTHONUNBUFFERED=1
 
-RUN [ ! -d "/datastore" ] && mkdir /datastore
-
 # Re #80, sets SECLEVEL=1 in openssl.conf to allow monitoring sites with weak/old cipher suites
 RUN sed -i 's/^CipherString = .*/CipherString = DEFAULT@SECLEVEL=1/' /etc/ssl/openssl.cnf
 
 # Copy modules over to the final image and add their dir to PYTHONPATH
 COPY --from=builder /dependencies /usr/local
-ENV PYTHONPATH=/usr/local
+ENV PYTHONPATH=/usr/local \
+    DATASTORE_DIR="/datastore"
 
 EXPOSE 5000
 
+# The entrypoint script handling PUID/PGID and permissions
+COPY docker-entrypoint.sh /app/docker-entrypoint.sh
+
 # The actual flask app
 COPY changedetectionio /app/changedetectionio
 
 # The eventlet server wrapper
 COPY changedetection.py /app/changedetection.py
 
+RUN chmod 777 /app/docker-entrypoint.sh && \
```

Comment:
why are we doing all this work then letting it be run and edited by anyone? `777` ?

Reply:
I have no answer to that. `755` is sufficient.